mesh/meshdb.go

```diff
@@ -489,6 +508,43 @@ func (m *DB) GetRewards(account types.Address) (rewards []types.Reward, err erro
 			Layer:               types.LayerID(layer),
 			TotalReward:         reward.TotalReward,
 			LayerRewardEstimate: reward.LayerRewardEstimate,
+			SmesherID:           reward.SmesherID,
+			Coinbase:            reward.Coinbase,
+		})
+	}
+	return
+}
+
+// GetRewardsBySmesherID retrieves rewards by smesherID
+func (m *DB) GetRewardsBySmesherID(smesherID types.NodeID) (rewards []types.Reward, err error) {
+	it := m.transactions.Find(getSmesherRewardKeyPrefix(smesherID))
+	for it.Next() {
+		if it.Key() == nil {
+			break
+		}
+		str := string(it.Key())
+		strs := strings.Split(str, "_")
+		layer, err := strconv.ParseUint(strs[2], 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("wrong key in db %s: %v", it.Key(), err)
+		}
+		//find the key to the actual reward, which is in it.Value()
+		var reward dbReward
+		rewardBytes, err := m.transactions.Get(it.Value())
+
+		if err != nil {
+			return nil, fmt.Errorf("wrong key in db %s: %v", it.Value(), err)
+		}
+		err = types.BytesToInterface(rewardBytes, &reward)
```

Comment:
You can combine this line and the next line using the `if err = foo; err != nil {` syntax